python/flip/main.cpp

```diff
@@ -0,0 +1,284 @@
+            float displayWidthPixels = py::cast<float>(vc[1]);
+            float displayWidthMeters = py::cast<float>(vc[2]);
+            parameters.PPD = FLIP::calculatePPD(distanceToDisplay, displayWidthPixels, displayWidthPixels);
+        }
+        else if (key == "startExposure")
+        {
+            parameters.startExposure = py::cast<float>(item.second);
+        }
+        else if (key == "stopExposure")
+        {
+            parameters.stopExposure = py::cast<float>(item.second);
+        }
+        else if (key == "numExposures")
+        {
+            parameters.numExposures = py::cast<int>(item.second);
+        }
+        else if (key == "tonemapper")
+        {
+            parameters.tonemapper = py::cast<std::string>(item.second);
+        }
+        else
+        {
+            std::cout << errorMessage << std::endl;
+        }
+    }
+
+    return parameters;
+}
+
+// Update parameter dictionary that is returned to the Python side.
+void updateInputParameters(const FLIP::Parameters& parameters, py::dict& inputParameters)
+{
+    inputParameters["ppd"] = parameters.PPD;
+    inputParameters["startExposure"] = parameters.startExposure;
+    inputParameters["stopExposure"] = parameters.stopExposure;
+    inputParameters["numExposures"] = parameters.numExposures;
+    inputParameters["tonemapper"] = parameters.tonemapper;
+}
+
+/** A simplified function for evaluating (the image metric called) FLIP between a reference image and a test image. Corresponds to the fourth evaluate() option in FLIP.h.
+*
+* @param[in] referenceInput Reference input image. For LDR, the content should be in [0,1]. The image is expected to have 3 floats per pixel.
+* @param[in] testInput Test input image. For LDR, the content should be in [0,1].
+* @param[in,out] parameters Contains parameters (e.g., PPD, exposure settings,etc). If the exposures have not been set by the user, then those will be computed (and returned).
+* @param[in] useHDR Set to true if the input images are to be considered containing HDR content, i.e., not necessarily in [0,1].
+* @param[in] applyMagmaMapToOutput A boolean indicating whether the output should have the MagmaMap applied to it before the image is returned.
+* @param[out] errorMapFLIPOutput The computed FLIP error image is returned in this variable. If applyMagmaMapToOutput is true, the function will allocate
+*             three channels (and store the magma-mapped FLIP images in sRGB), and
+*             if it is false, only one channel will be allocated (and the FLIP error is returned in that grayscale image).
+*             Note that the user is responsible for deallocating the errorMapFLIPOutput image.
+*/
+std::tuple<py::array_t<float>, float, py::dict> evaluate(const py::array_t<float> referenceInput, const py::array_t<float> testInput, const bool inputHDR, const bool inputsRGB = true, const bool returnMeanError = true, const bool applyMagma = true, py::dict inputParameters = {})
```

Comment:
`inputHDR` should be called `useHDR`?
Any reason why this function cannot return the meanFLIP, i.e., like the cpp-function?

Reply:
Fixed!